specs/plugin-extensibility.md

```diff
@@ -197,7 +187,19 @@ All response attributes are required.
 
 *supported-contract-versions* - The list of contract versions supported by the plugin. Currently this list must include only one version, per major version. Post initial release, Notation may add new features through plugins, in the form of new commands (e.g. tsa-sign for timestamping), or additional request and response parameters. Notation will publish updates to plugin interface along with appropriate contract version update. Backwards compatible changes (changes for which older version of plugin continue to work with versions of Notation using newer contract version) like new optional parameters on existing contracts, and new commands will be supported through minor version contract updates, breaking changes through major version updates. To maintain forward compatibility plugin implementors MUST ignore unrecognized attributes in command request which are introduced in minor version updates of the plugin contract. Plugin `get-plugin-metadata` command returns the contract version a plugin supports. Notation will evaluate the minimum plugin version required to satisfy a user's request, and reject the request if the plugin does not support the required version.
 
-*capabilities* - Non empty list of features supported by a plugin. Each capability such as `SIGNATURE_GENERATOR.RAW` requires one of more commands to be implemented by the plugin. When new features are available for plugins to implement, an implementation may choose to not implement it, and therefore will not include the feature in capabilities. Notation will evaluate the capability required to satisfy a user’s request, and reject the request if the plugin does not support the required capability.
+*capabilities* - A non-empty list of features supported by the plugin. Each capability requires the plugin to implement specific commands. Implementations of Notary Project spec can evaluate the required capability of a user's request and may reject the request if the plugin does not support it.
+
+Valid capabilities and required commands:
+  - `SIGNATURE_GENERATOR.RAW`: Generates a raw signature for OCI or blob payloads.
+    - Required commands: `describe-key`, `generate-signature`
+  - `SIGNATURE_GENERATOR.ENVELOPE`: Generates a complete signature envelope for OCI payloads.
+    - Required command: `generate-envelope`
+  - `SIGNATURE_GENERATOR.ENVELOPE_FOR_BLOB`: Generates a complete signature envelope for blob (local file) payloads.
```

Comment:
maybe we can remove `(local file)`, because it can be a blob stored in a storage in the future, then it is not a file, instead could be a reference to blob in the storage?

Reply:
Updated.